scripts/process/dv_stats.R

```diff
@@ -0,0 +1,57 @@
+#' @title Calculate the stat category for each gage's discharge value
+#' 
+#' @param viz a vizlab object that depends on a date, table of site stats, and table of discharge data
+process.dv_stats <- function(viz){
+  library(dplyr)
+  
+  deps <- readDepends(viz)
+  checkRequired(deps, c("date", "site_stats", "data"))
+  date <- deps[["date"]][["date"]]
+  site_stats <- deps[["site_stats"]]
+  data <- deps[["data"]]
+  
+  get_dv <- function(site_no){
+    out <- rep(NA, length(site_no))
+    for (i in 1:length(site_no)){
+      val <- data[[site_no[i]]][1L]
+      if (!is.null(val)){
+        out[i] <- val
+      }
+    }
+    return(out)
+  }
+  
+  int_per <- function(p05_va, p10_va, p20_va, p25_va, p50_va, p75_va, p80_va, p95_va, dv_val){
+    out <- rep(NA, length(dv_val))
+    
+    
+    for (i in 1:length(out)){
+      y <- c(0.05, 0.1, 0.2, 0.25, 0.5, 0.75, 0.8, 0.95)
+      x <- c(p05_va[i], p10_va[i], p20_va[i], p25_va[i], p50_va[i], p75_va[i], p80_va[i], p95_va[i]) %>% as.numeric
+      nas <- is.na(x)
+      x <- x[!nas]
+      y <- y[!nas]
+      if (length(unique(x)) < 2){
+        out[i] <- NA
+      } else if (dv_val[i] < x[1L]){ # the first and last *have* to be numbers per filtering criteria
+        out[i] <- 0.05
+      } else if (dv_val[i] > tail(x, 1L)){
+        out[i] <- 0.95
+      } else {
+        out[i] <- approx(x, y, xout = dv_val[i])$y
```

Comment:
This is a later decisionpoint, but noting it here. We'll want to decide if we want the output to be continuous or categorical for percentage. Here it is coming out as continuous, unless the dv is < 0.05 or > 0.95. Instead, we may want to assign the dv to a percentile category, since I don't think we'll want to display continuous color-space for these dots (or maybe we do?) and instead we'll have X number of percentile categories and those will be used for the color categories.

Reply:
We could use `cut` after `approx` to do this.

```
dv_val <- 380
x <- c(0.56, 0.94, 1.2, 1.4, 4.5, 45, 141, 448)
y <- c(0.05, 0.1, 0.2, 0.25, 0.5, 0.75, 0.8, 0.95)
per <- approx(x, y, xout = dv_val)$y
cut(per, y)
[1]  (0.8,0.95]
```

Reply:
Or use `cut` to assign colors later when plotting.

```
category_colors <- c("red", "darkred", "orange", "yellow", "green", "cyan", "skyblue", "blue", "black")
cut(per, c(0, y, 100), category_colors)
[1] blue
```

Reply:
yes, cut seems like a good option here.